pkg/kexec/memory_test.go

```diff
@@ -0,0 +1,261 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package kexec
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"reflect"
+	"testing"
+)
+
+func TestParseMemoryMap(t *testing.T) {
+	var mem Memory
+	root, err := ioutil.TempDir("", "memmap")
+	if err != nil {
+		t.Fatalf("Cannot create test dir: %v", err)
+	}
+	defer os.RemoveAll(root)
+
+	old := memoryMapRoot
+	memoryMapRoot = root
+	defer func() { memoryMapRoot = old }()
+
+	var want []TypedAddressRange
+	for i := 0; i < 4; i++ {
```

Comment:
Could you lay this out like a table driven test? Maybe just a struct?

```go
for i, file := range []struct{
  start uintptr
  size uintptr
  startstr string
  endstr string
  type string
}{
  os.Mkdir..
  ioutil.WriteFile(.., file.startstr, 0655)
  ioutil.WriteFile(..., file.sizestr, 0655)
  want = append(want, ...)
}
```

I'd prefer manual conversions for unit tests over generated ones in this case.

Reply:
Removed generation. One test covers all cases, so no need in a table driven test.